src/ln/channelmanager.rs

```diff
@@ -666,19 +689,21 @@ impl ChannelManager {
 
 	#[inline]
 	fn finish_force_close_channel(&self, shutdown_res: ShutdownResult) {
-		let (local_txn, mut failed_htlcs) = shutdown_res;
+		let (local_txn, mut failed_htlcs, claim_from_chain) = shutdown_res;
 		log_trace!(self, "Finishing force-closure of channel with {} transactions to broadcast and {} HTLCs to fail", local_txn.len(), failed_htlcs.len());
 		for htlc_source in failed_htlcs.drain(..) {
 			self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), htlc_source.0, &htlc_source.1, HTLCFailReason::Reason { failure_code: 0x4000 | 8, data: Vec::new() });
 		}
-		for tx in local_txn {
-			self.tx_broadcaster.broadcast_transaction(&tx);
-		}
+		if claim_from_chain {
+			for tx in local_txn {
+				self.tx_broadcaster.broadcast_transaction(&tx);
+			}
+		};
 	}
 
 	/// Force closes a channel, immediately broadcasting the latest local commitment transaction to
 	/// the chain and rejecting new HTLCs on the given channel.
-	pub fn force_close_channel(&self, channel_id: &[u8; 32]) {
+	pub fn force_close_channel(&self, channel_id: &[u8; 32], store_closed : bool) {
```

Comment:
Please document the new public parameter and what it does (also, "store_closed" is a weird term for this, maybe "delay_tx_broadcast" or so?